packages/server/src/Document.ts

```diff
@@ -215,6 +215,15 @@ export class Document extends Doc {
 
     return this
   }
+
+  /**
+   * Broadcast stateless message to all connections
+   */
+  public sendStateless(payload: string): void {
```

Comment:
It would be nice to have an optional argument here to send a message to a specific connection instead of all connections.

That would support the simple PING/PONG use case, as well as others where a single connection wants to message back and forth with the server.

I suppose if `connection.sendStateless` is a public method though, that this is already possible?

Reply:
If you want to send a message to a specific connection, you can search for the corresponding connection from the `document.connections` and then call `connection.sendStateless`.

And then, if you want to reply to a message sent by a certain provider, you can use the `connection` parameter in the onStateless hook.

I add some test cases.